docs/source/tools/speech_data_processor.rst

```diff
@@ -5,4 +5,134 @@ Speech Data Processor (SDP) is a toolkit to make it easy to:
   1. write code to process a new dataset, minimizing the amount of boilerplate code required.
   2. share the steps for processing a speech dataset. Sharing processing steps can be as easy as sharing a YAML file.
 
-SDP is hosted here: https://github.com/NVIDIA/NeMo-speech-data-processor. 
\ No newline at end of file
+SDP is hosted here: https://github.com/NVIDIA/NeMo-speech-data-processor. 
+
+SDP's philosophy is to represent processing operations as 'processor' classes. Many common processing operations are provided, and it is easy to add your own. In some cases, all you will need to do to process a new dataset is simply to write a YAML file containing the parameters needed to process your dataset.
+
+SDP is specifically intended for the use case when you have an existing dataset with the audio & text pairs already specified in some form, and you wish to create a JSON manifest suitable for use with NeMo. SDP allows for intermediate cleaning and filtering steps which involve amending the 'ground truth' ``"text"`` or dropping utterances which are deemed to be too inaccurate for training on.
+
+Overview of how SDP processes a dataset
+---------------------------------------
+
+1. You call the ``main.py`` script, passing in a YAML config file, possibly with some overrides.
+2. ``main.py`` script calls ``run_processors.py``, passing in your config.
+3. ``run_processors.py`` does the following:
+
+a. picks out the processors you wish to run (you can specify a subset of the processors in the config override, e.g. to avoid re-running time-consuming steps).
+b. if some of the processors have not had "output_manifest_file" or "input_manfiest_file" entries specified, SDP will automatically create temporary files for those.
```

Comment:
Perhaps some explanation of what these args mean in a manifest before detailing they will be created automatically.